problem-solver/py/Dockerfile

```diff
@@ -0,0 +1,7 @@
+#Deriving the latest base image
+FROM python:latest
+
+COPY . ./
+RUN pip install --requirement ./requirements.txt
+
+ENTRYPOINT ["python", "./server.py"]
```

Comment:
```suggestion
FROM python:3.11-slim as base

USER root
WORKDIR /nika/problem-solver/py

RUN apt-get update && apt-get install -y --no-install-recommends tini

COPY ./problem-solver/py/requirements.txt .
RUN --mount=type=cache,target=/root/.cache/pip pip3 install -r requirements.txt

COPY ./problem-solver/py/ .
COPY scripts/run_py_sc_server.sh scripts/set_vars.sh /nika/scripts/

ENTRYPOINT [ "/usr/bin/tini", "--", "./scripts/run_py_sc_server.sh" ]
```
Suggested changes:
1. pinning a python version would allow this image to be built deterministically (e.g. with the `latest` tag, the build may fail due to incompatibility of future version of python with the dependencies locked by `requirements.txt`)
2. `slim` version will reduce the size of the image
3. To avoid logic duplication and inconsistencies we should either delete the `run_py_sc_server.sh` script or use it inside Docker as well. In my suggestion I've reused the script, but for that `docker-compose.yml`'s build syntax has to be changed. 
4. `tini` is used in conjunction with the proposed usage of the script to speed up container termination (eliminating the 10s wait time)
5. BuildKit build-time mount for pip would allow building images faster even if `requirements.txt` changes, that might be helpful for testing while developing new agents.
6. requirements.txt is copied before the rest of the source files to utilize Docker build stage caching if `requirements.txt` hasn't changed